python/sdk/setup.py

```diff
@@ -31,7 +31,7 @@
     "docker>=4.2.1",
     "google-auth>=1.11.0",
     "google-cloud-storage>=1.19.0",
-    "mlflow>=1.2.0,<=1.23.0", #  for py3.11 due to proto -> "mlflow>=1.26.1",
+    "mlflow>=1.23.0",
```

Comment:
I think it will break compatibility with the user's model created using `mlflow` < 1.23.0

Reply:
@ariefrahmansyah removing this constraint also requires some code changes in the pyfunc server to properly handle the compatibility. @tiopramayudi know more details about this.

Reply:
removing the constraint will become problematic if user specified the conda dependency lower than `mlflow` < 1.23.0 it ill failed when mlflow load the model from artifact due to different number of arguments passed

Reply:
Got it. Thanks for explaining it to me 🙏🏻 

More context on why I removed the MLflow constraint, I was using mlflow==1.30.0 in my machine when working on this PR and it was working. I can run the pyfunc server, locally and remotely in the Merlin server. But after Tio's explanation, I realized that my env.yaml does not pin the mlflow version which allows MLflow to add mlflow without constraint in the logged model's conda.yaml.

Moving forward, I would like to remove this hard limit on MLflow version 🤞🏻

Reply:
@ariefrahmansyah @tiopramayudi How complex is the fix?

Reply:
I'm still not sure. We might need to do some regression tests. However, I don't think we can maintain backward compatibility here.

I have failed to run the guide in mlflow==1.23.0. But it succeeds with mlflow==1.30.0 using the same code and commands

With mlflow 1.23.0
```
➜ pip freeze | grep mlflow
mlflow==1.23.0

➜ python examples/iris-model/log_local.py
[16:59:39] WARNING: /Users/runner/work/xgboost/xgboost/python-package/build/temp.macosx-10.9-x86_64-cpython-37/xgboost/src/learner.cc:627:
Parameters: { "silent" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.


PyFunc model have logged locally.
To run PyFunc Server locally, execute this command:

	python -m pyfuncserver --model_dir file:///Users/ariefrahmansyah/go/src/github.com/gojek/merlin/python/pyfunc-server/mlruns/0/c775f8e2ae3049ac857b173aef43fec8/artifacts/model

➜ python -m pyfuncserver --model_dir file:///Users/ariefrahmansyah/go/src/github.com/gojek/merlin/python/pyfunc-server/mlruns/0/c775f8e2ae3049ac857b173aef43fec8/artifacts/model
ERROR:root:Unable to initalize model
ERROR:root:Traceback (most recent call last):
  File "/Users/ariefrahmansyah/go/src/github.com/gojek/merlin/python/pyfunc-server/pyfuncserver/__main__.py", line 47, in <module>
    model.load()
  File "/Users/ariefrahmansyah/go/src/github.com/gojek/merlin/python/pyfunc-server/pyfuncserver/model/model.py", line 62, in load
    self._model = pyfunc.load_model(self.artifact_dir)
  File "/Users/ariefrahmansyah/.pyenv/versions/miniconda3-3.7-4.11.0/envs/local-pyfunc/lib/python3.7/site-packages/mlflow/pyfunc/__init__.py", line 667, in load_model
    model_impl = importlib.import_module(conf[MAIN])._load_pyfunc(data_path)
  File "/Users/ariefrahmansyah/.pyenv/versions/miniconda3-3.7-4.11.0/envs/local-pyfunc/lib/python3.7/site-packages/mlflow/pyfunc/model.py", line 269, in _load_pyfunc
    python_model = cloudpickle.load(f)
ModuleNotFoundError: No module named 'iris_model'

ERROR:root:
Some module might be missing from your pyfunc model.
If the module is part of your code,
you can specify it as 'code' argument when calling log_pyfunc, e.g.:

merlin.log_pyfunc_model(model_instance=MyModel(),
        conda_env="env.yaml",
        code=["path/to/your/package"])

If the module is a python package, then you should add it in the
conda env.yaml that you pass to log_pyfunc method, e.g.:

dependencies:
- python=3.7
- pip:
  - my-missing-package==1.0

OR if the package is not available in PyPI distribution

dependencies:
- python=3.7
- my-missing-package=1.0

See: https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#creating-an-environment-from-an-environment-yml-file
```

With mlflow 1.30.0
```
➜ pip freeze | grep mlflow
mlflow==1.30.0

➜ python examples/iris-model/log_local.py
[16:58:16] WARNING: /Users/runner/work/xgboost/xgboost/python-package/build/temp.macosx-10.9-x86_64-cpython-37/xgboost/src/learner.cc:627:
Parameters: { "silent" } might not be used.

  This could be a false alarm, with some parameters getting used by language bindings but
  then being mistakenly passed down to XGBoost core, or some parameter actually being used
  but getting flagged wrongly here. Please open an issue if you find any such cases.


/Users/ariefrahmansyah/.pyenv/versions/miniconda3-3.7-4.11.0/envs/local-pyfunc/lib/python3.7/site-packages/_distutils_hack/__init__.py:33: UserWarning: Setuptools is replacing distutils.
  warnings.warn("Setuptools is replacing distutils.")
PyFunc model have logged locally.
To run PyFunc Server locally, execute this command:

	python -m pyfuncserver --model_dir file:///Users/ariefrahmansyah/go/src/github.com/gojek/merlin/python/pyfunc-server/mlruns/0/ac0e10cff18949fcb1ec2bb6f44c5fe5/artifacts/model

➜ python -m pyfuncserver --model_dir file:///Users/ariefrahmansyah/go/src/github.com/gojek/merlin/python/pyfunc-server/mlruns/0/ac0e10cff18949fcb1ec2bb6f44c5fe5/artifacts/model
2023/02/27 16:58:30 WARNING mlflow.pyfunc: Detected one or more mismatches between the model's dependencies and the current Python environment:
 - mlflow (current: 1.30.0, required: mlflow==1.23.0)
To fix the mismatches, call `mlflow.pyfunc.get_model_dependencies(model_uri)` to fetch the model's environment and install dependencies using the resulting environment file.
INFO:root:Registering model: name: model, version: 1, fullname: model-1
INFO:root:Listening on port 8080
INFO:root:Will fork 1 workers
^CINFO:root:Stop accepting new connection
INFO:root:Closing existing connection
INFO:root:Stopping ioloop
```

Reply:
The output of mlruns are similar though
![image](https://user-images.githubusercontent.com/8122852/221534953-ebd514bb-49de-48cd-a541-b11033c0c113.png)
![image](https://user-images.githubusercontent.com/8122852/221534987-16127ccb-b315-4fd7-a08d-31bff915e963.png)